source/test/TestDirector.cc

```diff
@@ -159,12 +175,24 @@ void TestDirector::writeTestOutput() const {
 TestData TestDirector::findNextEntry() {
     EGG::Vector3f pos;
     EGG::Quatf fullRot;
+    EGG::Vector3f extVel;
+    EGG::Vector3f intVel;
     pos.read(m_stream);
     fullRot.read(m_stream);
 
+    if (m_versionMinor >= ADDED_EXT_VEL) {
+        extVel.read(m_stream);
+    }
+
+    if (m_versionMinor >= ADDED_INT_VEL) {
+        intVel.read(m_stream);
+    }
```

Comment:
IF we want to try to use the same switch approach here, then we actually need to add NEW datapoints to the BEGINNING of the frame's data in the KRKG. See the below example, which would read in each value in reverse order

```
switch(m_versionMinor) {

case ADDED_INT_VEL:
    intVel.read(m_stream);
case ADDED_EXT_VEL:
    extVel.read(m_stream);
default:
    pos.read(m_stream);
    fullRot.read(m_stream);
}
```